rfd/0062-tctl-sso-configure-command.md

```diff
@@ -0,0 +1,205 @@
+endpoint.
+
+Additionally, the command provides a foundation for creating similar functionality in web-based UI.
+
+## UX
+
+Each connector kind is served by a separate subcommand. Most subcommand flags are not shared.
+
+```bash
+$ tctl sso configure --help
+
+Create auth connector configuration.
+
+...
+
+Commands:
+  sso configure saml   Configure SAML connector, optionally using a preset. Available presets: [okta onelogin ad adfs]
+  sso configure oidc   Configure OIDC auth connector, optionally using a preset.
+  sso configure github Configure GitHub auth connector.
+```
+
+### Flags: SAML
+
+For `SAML` the commonly used flags will be:
+
+```
+-p, --preset                   Preset. One of: [okta onelogin ad adfs]
+-n, --name                     Connector name. Required, unless implied from preset.
+-e, --entity_descriptor        Set the Entity Descriptor. Valid values: file, URL, XML content. Supplies configuration parameters as single XML instead of individual elements.
+-a, --attributes_to_roles      Sets attribute-to-role mapping in the form 'attr_name,attr_value,role1,role2,...'. Repeatable.
+    --display                  Display controls how this connector is displayed.
+```
+
+The `--attributes_to_roles/-a` flag is particularly important as it is used to provide a mapping between the
+IdP-provided attributes and Teleport roles. It can be specified multiple times.
+
+Alternatives to `--entity_descriptor/-e` flag, allowing to specify these values explicitly:
+
+```
+--issuer                   Issuer is the identity provider issuer.
+--sso                      SSO is the URL of the identity provider's SSO service.
+--cert                     Cert is the identity provider certificate PEM. IDP signs <Response> responses using this certificate.
+--cert_file                Like --cert, but read the cert from file.
+```
+
+Rarely used flags; the tool fills that information automatically:
+
+```
+--acs                      AssertionConsumerService is a URL for assertion consumer service on the service provider (Teleport's side).
+--audience                 Audience uniquely identifies our service provider.
+--service_provider_issuer  ServiceProviderIssuer is the issuer of the service provider (Teleport).
+```
+
+Advanced features:
+
+- assertion encryption (support varies by IdP)
+- externally-provided signing key (by default Teleport will self-issue this)
+- overrides for specific IdP providers
+
+```
+--signing_key_file         A file with request signing key. Must be used together with --signing_cert_file.
+--signing_cert_file        A file with request certificate. Must be used together with --signing_key_file.
+--assertion_key_file       A file with key used for securing SAML assertions. Must be used together with --assertion_cert_file.
+--assertion_cert_file      A file with cert used for securing SAML assertions. Must be used together with --assertion_key_file.
```

Comment:
I'm pretty sure that our CLI flags use dashes instead of underscores so I'd rename all these new flags to match. I left the same comment in your implementation PR.

Reply:
Thanks, done.